cli/tests/extensions.rs

```diff
@@ -280,6 +280,135 @@ fn conflicting_extension_name_is_filtered() {
     assert!(output.contains("extension was filtered out"));
 }
 
+mod permissions {
```

Comment:
This file is hundreds of lines long already anyway, why not move this into an actual separate module?